video_xblock/static/js/player-context-menu.js

```diff
@@ -16,82 +19,93 @@ domReady(function() {
   // VideoJS Player() object necessary for context menu creation
   var player = videojs('{{ video_player_id }}');
 
+  /**
+   * Create elements of nested context submenu.
+   */
   function createNestedContextSubMenu(e) {
     var target = e.target;
 
     if (target.matches("li.vjs-menu-item")
         && target.innerText == player.contextmenuUI.content[3].label
         && !target.querySelector('.vjs-contextmenu-ui-submenu') ) {
-
       // Create nested submenu
       var ulSubMenu = document.createElement('ul');
       ulSubMenu.className = 'vjs-contextmenu-ui-submenu';
       target.appendChild(ulSubMenu);
       for (var i = 0; i < playbackRates.length; i++) {
-        var liSubMenu = document.createElement('li');
-        liSubMenu.className = 'vjs-submenu-item';
-        liSubMenu.innerHTML = playbackRates[i] + 'x';
-        ulSubMenu.appendChild(liSubMenu);
-        liSubMenu.onclick = function(){
-          player.playbackRate(parseFloat(this.innerHTML));
-        }
+        (function() {
+          var liSubMenu = document.createElement('li');
+          var playbackRate = playbackRates[i];
+          liSubMenu.className = 'vjs-submenu-item';
+          liSubMenu.innerHTML = playbackRate + 'x';
+          ulSubMenu.appendChild(liSubMenu);
+          liSubMenu.onclick = function() {
+            player.playbackRate(parseFloat(playbackRate));
+          }
+        })();
```

Comment:
it's better to use DocumentFragment in bulk DOM elements generations.
`forEach` method can also help you to simply the code.